internal/attestation/attestation_test.go

```diff
@@ -64,22 +52,71 @@ func TestTLSConfig(t *testing.T) {
 		return nil
 	}
 
-	clientConfig := CreateAttestationClientTLSConfig(verifyRemoteReport, verifyReport)
+	failToVerifyReport := func(report attestation.Report) error {
+		return errors.New("invalid report")
+	}
+
+	//
+	// Create Test Cases
+	//
 
-	client := http.Client{Transport: &http.Transport{TLSClientConfig: clientConfig}}
+	tests := []struct {
+		name               string
+		getRemoteReport    interface{}
+		verifyRemoteReport interface{}
+		verifyReport       interface{}
+		expectErr          bool
+	}{
+		{"basic", getRemoteReport, verifyRemoteReport, verifyReport, false},
+		{"invalid remote report", getRemoteReport, failToVerifyRemoteReport, verifyReport, true},
+		{"invalid report", getRemoteReport, verifyRemoteReport, failToVerifyReport, true},
+		{"invalid remote report and report", getRemoteReport, failToVerifyRemoteReport, failToVerifyReport, true},
+	}
 
 	//
-	// Test connection.
+	// Run Tests.
 	//
 
-	server.StartTLS()
-	defer server.Close()
+	for _, test := range tests {
+		t.Logf("Subtest: %v", test.name)
+		//
+		// Create server.
+		//
+
+		serverConfig, err := CreateAttestationServerTLSConfig(test.getRemoteReport.(func([]byte) ([]byte, error)))
+		require.NoError(err)
+
+		server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			io.WriteString(w, "hello")
+		}))
+		server.TLS = serverConfig
+
+		//
+		// Create client.
+		//
+
+		clientConfig := CreateAttestationClientTLSConfig(test.verifyRemoteReport.(func([]byte) (attestation.Report, error)), test.verifyReport.(func(attestation.Report) error))
+		client := http.Client{Transport: &http.Transport{TLSClientConfig: clientConfig}}
 
-	resp, err := client.Get(server.URL)
-	require.NoError(err)
-	defer resp.Body.Close()
+		//
+		// Test connection.
+		//
 
-	body, err := ioutil.ReadAll(resp.Body)
-	require.NoError(err)
-	assert.EqualValues("hello", body)
+		server.StartTLS()
+
+		resp, err := client.Get(server.URL)
+		if test.expectErr {
+			require.Error(err)
+			continue
+		} else {
+			require.NoError(err)
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(err)
+		assert.EqualValues("hello", body)
+
+		server.Close()
+		resp.Body.Close()
```

Comment:
This code has multiple exits where things don't get closed, so I suggest the following pattern:
```suggestion
		func() {
			server.StartTLS()
			defer server.Close()

			resp, err := client.Get(server.URL)
			if test.expectErr {
				require.Error(err)
				return
			}
			require.NoError(err)
			defer resp.Body.Close()

			body, err := ioutil.ReadAll(resp.Body)
			require.NoError(err)
			assert.EqualValues("hello", body)
		}()
```